core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -383,6 +384,16 @@ public FileInfo getFileInfo(AlluxioURI path)
       mPermissionChecker.checkPermission(FileSystemAction.READ, path);
       // getFileInfo should load from ufs if the file does not exist
       getFileId(path);
+
+      if (!mInodeTree.inodePathExists(path)) {
```

Comment:
It is slightly more efficient. It avoids resolve the path in ufs and ufs.exists, ufs.isFile calls. I can move that inside loadMetadata if you prefer